### ISSUE TITLE
Generate a subset of models based on convenience methods specified in a Cadl file

### ISSUE THREAD

**Issue by annelo-msft** [Azure/autorest.csharp#2336]
This tracks the work to generate input and output models for operations listed in autorest.md.  It tracks creation of the mechanism to specify the operation in the autorest.md file and from it determine the list of models, not to generate those models.  Generation of the models themselves will be tracked in a separate issue.

The input describing the API is Cadl.

**Comment by annelo-msft** [comment]
Pending https://github.com/Azure/cadl-azure/issues/1839

**Comment by lirenhe** [comment]
based on the discussion with @lmazuel, let's use 

@OpenAPI.extension("x-ms-convenient-api", true)
op foo(): {};

As a workaround to unblock this work.  Please note, the JSON/YAML that comes out of the emitter must be designed correctly from the start,

**Comment by lirenhe** [comment]
We shall also filter the models for convinience methods instead of generating a full list of models.

**Comment by chunyu3** [comment]
depend on https://github.com/microsoft/cadl/issues/1068,
https://github.com/microsoft/cadl/issues/1046

**Comment by chunyu3** [comment]
Wait for the cadl compiler and rest new version to include the resolve of issue 1068 and 1046
